uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-actions.component.js

```diff
@@ -197,6 +185,41 @@ define([
       return allowCancellationValue === '2';
     }
 
+    /**
+     * Opens dialog and immediately starts checking balance change
+     * If balance changed, prompts to recalculate the balance change,
+     * if not - simply asks for the action confirmation
+     *
+     * @param {String} action
+     */
+    function checkBalanceChangeAndPromptForAnAction (action) {
+      dialog.open({
+        title: 'Checking balance change...',
+        loading: true,
+        delayedProps: function () {
+          return vm.leaveRequest.checkIfBalanceChangeNeedsForceRecalculation()
+            .then(function (balanceChangeNeedsForceRecalculation) {
+              if (balanceChangeNeedsForceRecalculation) {
+                return _.assign(
+                  LeaveRequestService.getBalanceChangeRecalculationPromptOptions(),
```

Comment:
I'm kind of worried that we are using `delayedProps` to also execute some code that at the end won't change the dialog properties, and not because we shouldn't do this, but because of the name. Maybe we should name this `loadingPromise` and to override the scope it should be something like this:

```js
dialog.open({
  title: 'Loading...',
  loadingPromise: function ($scope) {
    return $q.resolve().then(function () {
      $scope.title = 'Done!';
      // or
      _.assign($scope, { values });
    });
  }
});
```

Reply:
Renamed